src/tools/tools.h

```diff
@@ -9,11 +9,12 @@ struct frame_data;
 
 struct tool
 {
-    virtual ~tool() {}
+    virtual ~tool() = default;
 
     virtual void use(raycast_info *rc) = 0;
     virtual void alt_use(raycast_info *rc) = 0;
     virtual void long_use(raycast_info *rc) = 0;
+    virtual void long_alt_use(raycast_info *rc) {}
```

Comment:
Does doing the raycast outside the tool still make sense?

Reply:
I think so. Otherwise most (every?) tool will have that same bit of code, but need access to player.